packages/providers/openai/src/live.ts

```diff
@@ -0,0 +1,164 @@
+import { LiveLLM, type LiveConnectConfig } from "@llamaindex/core/llms";
+import { getEnv } from "@llamaindex/env";
+import type { ChatModel } from "openai/resources.mjs";
+import { OpenAILiveSession } from "./live-session";
+import { OpenAI } from "./llm";
+import {
+  mapModalityToOpenAIModality,
+  type OpenAILiveConfig,
+  type OpenAIVoiceNames,
+} from "./utils";
+
+export class OpenAILive extends LiveLLM {
+  private apiKey: string | undefined;
+  private model: ChatModel | (string & {});
+  voiceName?: OpenAIVoiceNames | undefined;
+  private baseURL: string;
+
+  constructor(init?: OpenAILiveConfig) {
+    super();
+    this.apiKey = init?.apiKey ?? getEnv("OPENAI_API_KEY");
+    this.model = init?.model ?? "gpt-4o-realtime-preview-2025-06-03";
+    this.voiceName = init?.voiceName;
+    this.baseURL = "https://api.openai.com/v1/realtime";
+    if (!this.apiKey) {
+      throw new Error("OPENAI_API_KEY is not set");
+    }
+  }
+
+  private async getEPHEMERALKey() {
+    const response = await fetch(`${this.baseURL}/sessions`, {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer ${this.apiKey}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        model: this.model,
+        voice: this.voiceName,
+      }),
+    });
+
+    const data = await response.json();
+    return data.client_secret.value;
+  }
+
+  private async getSDPResponse(offer: RTCSessionDescriptionInit) {
+    const EPHEMERAL_KEY = await this.getEPHEMERALKey();
```

Comment:
needs to use apiKey (which will be the ephemeral one retrieved from the server if this is called on the client)